### PR TITLE
Validate create image name

### PR DESCRIPTION
**Pull request by resoluteCoder**
Added name validation for image with regex same as front end

### CODE REVIEW

File: pkg/models/images.go [hunk 1]
```diff
@@ -26,6 +27,8 @@ const (
 	DistributionCantBeNilMessage = "distribution can't be empty"
 	// ArchitectureCantBeEmptyMessage is the error message when the architecture is empty
 	ArchitectureCantBeEmptyMessage = "architecture can't be empty"
+	// NameCantBeInvalidMessage is the error message when the name is invalid
+	NameCantBeInvalidMessage = "invalid input for name"
```
Comment (boaz0) [inline pkg/models/images.go:31]: How about "name must start with alphanumeric characters and can contain underscore and hyphen characters"

File: pkg/models/images.go [hunk 2]
```diff
@@ -47,6 +50,9 @@ func (i *Image) ValidateRequest() error {
 	if i.Distribution == "" {
 		return errors.New(DistributionCantBeNilMessage)
 	}
+	if re := regexp.MustCompile(`^[A-Za-z0-9]+[A-Za-z0-9\s_-]*$`).FindString(i.Name); re == "" {
```
Comment (boaz0) [inline pkg/models/images.go:53]: I would rather do something else than `regexp` but I guess `regexp` is much cleaner.
Anyway, why not using `MatchString(i.Name)`, that way you don't need `re`:
```go
 	if regexp.MustCompile(`^[A-Za-z0-9]+[A-Za-z0-9\s_-]*$`).MatchString(i.Name) {
```

**Review by boaz0 — CHANGES_REQUESTED** [review]
Please add unit tests as well!

File: pkg/models/images.go (hunk 2 above)
Comment (resoluteCoder) [reply pkg/models/images.go:53]: With FindString I can get a value back that can be checked for false.

Well, I suppose I can write `MatchString(i.Name) == false` or 
` if !regexp.MustCompile(pattern).MatchString(i.Name) {` and adding the `!`

What do you think?

Comment (boaz0) [reply pkg/models/images.go]: Yes that's what I meant.

**Review by biancarosa — APPROVED** [review]

**Pull request merged**